### ISSUE TITLE
Feature request: app widgets

### ISSUE THREAD

**Issue by renyuneyun** [SubhamTyagi/Last-Launcher#7]
**Requested feature**:

Support app widgets.

The easiest approach may be to have the widgets exist on a different screen (e.g. swipe left or right).

**Explanation**:

This may be a long-term thing. I don't have much knowledge of developing launchers, but I can imagine app widgets are not just texts -- meaning they will be more difficult to handle.
The fact that they are AppWidget makes them more difficult to deal with -- I ran across this when developing the widgets of Easer.

But to be honest, I think app widget is one of the most important productivity features of Android's main screen (i.e. launcher). To my mind, it's really a shame that a launcher doesn't support app widgets (in its final form / version).

**Comment by tkarakashian** [comment]
Seconding this one.  I know it's listed as a feature of the launcher that it doesn't support widgets, but it's the only significant downside for me right now.

**Comment by c-o-m-m-a-n-d-e-r-d-a-t-a** [comment]
+1
Understand that graphical widgets would be impossible, but others like Greenify's sleep and hibernate are kind of essential for me.
Thanks.
This launcher is an awesome concept!

**Comment by SubhamTyagi** [comment]
App Widget support may be supported in later version (v2).
A swipe gesture (left or right but only one not both) will leads to new screen where App widget may be shown.

**Comment by ralmcruz** [comment]
This is a great launcher already. The absence of "icon and widgets" is a advantage for an launcher that is "simple and minimal".

**Comment by c-o-m-m-a-n-d-e-r-d-a-t-a** [comment]
> This is a great launcher already. The absence of "icon and widgets" is a advantage for an launcher that is "simple and minimal".

Sure, absolutely agree that it's a great launcher. It wouldn't be spoilt by having entries that were a functional task (rather than graphical). They could be made indistinguishable from the existing app entries, and it wouldn't alter the look and feel.

**Comment by renyuneyun** [comment]
@ralmcruz I agree this is a great launcher and the absence of icons on the main page is its advantage.
But it doesn't harm anything if the launcher "can" support widgets. I also felt it would spoil the feeling of the main page, and that's why I suggested adding it to a separated page instead of mixing into the main page.


------
Slightly modified the wording of the issue -- the last sentence was a bit offensive previously. Hope the author didn't mind :p

**Comment by QuibblingAsh42** [comment]
Agreed.  It's possible for two people to have the same wants or need at the macro level, but those differ in nuance.  I love Last's text-based interface, but wouldn't begrudge someone's desire to also have widgets as well.  It's also possible to have the capability of using widgets in the launcher and not USE that capability in practice.

**Comment by mubashir-rehman** [comment]
@renyuneyun I would suggest adding widgets at the bottom/top/left edge/right edge/ all sides of the home screen instead of whole new page. It kinda feels cluttering in thinking too but might be a nice addition.
This search query might help https://github.com/Neamar/KISS/search?q=widgets&type=commits .
